module/VuFind/src/VuFind/Db/Service/FeedbackService.php

```diff
@@ -50,4 +66,140 @@ public function createEntity(): Feedback
+
+        if (null !== $formName) {
+            $dqlWhere[] = "f.formName = :formName";
+            $parameters['formName'] = $formName;
+        }
+        if (null !== $siteUrl) {
+            $dqlWhere[] = "f.siteUrl = :siteUrl";
+            $parameters['siteUrl'] = $siteUrl;
+        }
+        if (null !== $status) {
+            $dqlWhere[] = "f.status = :status";
+            $parameters['status'] = $status;
+        }
+        if (!empty($dqlWhere)) {
+            $dql .= ' WHERE ' . implode(' AND ', $dqlWhere);
+        }
+        $dql .= " ORDER BY f.created DESC";
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters($parameters);
+
+        if (null !== $page) {
+            $query->setMaxResults($limit);
+            $query->setFirstResult($limit * ($page - 1));
+        }
+        $paginator = new Paginator($query);
+        $paginator->setUseOutputWalkers(false);
+        return $paginator;
+    }
+
+    /**
+     * Delete feedback by ids
+     *
+     * @param array $ids IDs
+     *
+     * @return int Count of deleted rows
+     */
+    public function deleteByIdArray(array $ids): int
+    {
+        // Do nothing if we have no IDs to delete!
+        if (empty($ids)) {
+            return 0;
+        }
+        $dql = 'DELETE FROM ' . $this->getEntityClass(Feedback::class) . ' fb '
+            . ' WHERE fb.id IN (:ids)';
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters(compact('ids'));
+        $query->execute();
+        return count($ids);
+    }
+
+    /**
+     * Get values for a column
+     *
+     * @param string $column Column name
+     *
+     * @return array
+     */
+    public function getColumn(string $column): array
+    {
+        $parameters = [];
+        $dql = "SELECT f.id, f." . $this->mapper($column)
+            . " FROM " . $this->getEntityClass(Feedback::class) . " f "
+            . "ORDER BY f." . $this->mapper($column);
+        $query = $this->entityManager->createQuery($dql);
+        return $query->getResult();
+    }
+
+    /**
+     * Update a column
+     *
+     * @param string $column Column name
+     * @param mixed  $value  Column value
+     * @param int    $id     id value
+     *
+     * @return bool
+     */
+    public function updateColumn($column, $value, $id)
+    {
+        $dql = "UPDATE " . $this->getEntityClass(Feedback::class) . " f "
+            . "SET f." . $this->mapper($column) . " = :value "
+            . " WHERE f.id = :id";
+        $parameters['value'] = $value;
+        $parameters['id'] = $id;
+        $query = $this->entityManager->createQuery($dql);
+        $query->setParameters($parameters);
+        return $query->execute();
+    }
+
+    /**
+     * Db columnn name to Doctrine entity field mapper
+     *
+     * @param string $column Column name
+     *
+     * @return string
+     */
+    public function mapper($column)
+    {
+        $map = [
```

Comment:
It might make sense to move this map to a class property so it only gets defined once instead of on every call.